Isaac/pipeline

isaac-gs opened the pull request:
Added a way to run the scikit algorithms from start to finish on the training data.

asaar committed:
Update README.md

isaac-gs committed:
Building out launcher (incomplete)

We need a pipeline that takes data and outputs results, I'm trying to
flesh out that pipeline. This code isn't nearly done yet.

isaac-gs committed:
Pipeline for training

The pipeline for training and viewing potential results when tested
against training data has a first draft.

isaac-gs committed:
Merging master with pipeline branch

isaac-gs committed:
Formatting fixes

ASAAR commented:
@RyanMcBerg @asclines Can one of you review?

isaac-gs committed:
Adding tests framework tools

asaar committed:
Create readme.MD

asclines commented on launcher.py:
```diff
@@ -0,0 +1,40 @@
+    if type == "train":
+        get_potential_results(training_file_name, separator_type, prediction_type)
+    else:
+        return # for applying on testing data, will need to both train the models and apply the trained models.
```
Can we get some clarification on what happens if the option is "test"?

ASAAR replied:
For that, it will train the models and then run them against the test data, making predictions.

asclines commented on modules/toolbox/framework_tools.py:
```diff
@@ -21,3 +20,44 @@ def get_train_test(path, filename, separator):
+    return pd.DataFrame.as_matrix(pandas_data)
+
+
+# not positive that this is a good enough check. TODO: Test thoroughly
```
How will this be tested? Should testing be done before PR?

ASAAR replied:
Uhhhh, yeah I'll work on that. You mean line 46 right?

asclines replied:
Yes

ASAAR replied:
@asclines So testing does exist, but the classification recognizer isn't working

asclines commented on modules/toolbox/ml_runners.py:
```diff
@@ -0,0 +1,19 @@
+# This file has functions which call different machine learning algorithms and handle results.
+import sys
+import os
+sys.path.insert(0, os.path.abspath('../..'))
```
Is this a sure thing?

ASAAR replied:
Yes, it's setup for each file individually. Not a pretty solution but I couldn't find a better one.

asclines commented on modules/toolbox/ml_runners.py:
```diff
@@ -0,0 +1,19 @@
+            print(item(path, filename, separator))
+
+
+def run_classifications():
```
Does this do anything?

ASAAR replied:
Not yet, I was waiting for Zak's code. It's up next for implementation

asclines reviewed (approved):
Overall good, as these tests get bigger have you considered rolling out or bringing in a logging system for debugging and whatnot?

isaac-gs committed:
Fixed issue of not being able to select target column

We still need to fix some of our tests because we don't have data
pre-processing.

isaac-gs committed:
Fixing function that identifies classification vs regression
